Police-Data-Accessibility-Project/data-sources-app-v2
# Update `/data-requests` `PUT` logic and tests

### Fixes

* https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/504

### Description

* Update `/data-requests/{resource_id}` `PUT` logic and tests
* Tests now test for ALL parameters that can be provided in the endpoint.
* Backend logic has been modified to account for cases where expected behavior was previously not occurring. 

### Testing

* Run tests in `tests` directory
* Inspect API documentation to confirm expected appearance.

### Performance

* Performance impacts due to increased database calls
* Increased test overhead.

### Docs

* API documentation updated

### Breaking Changes

* No breaking changes.
